tasks/setup.py

```diff
@@ -134,36 +135,24 @@ def setupAWSConfig(config: Config):
             break
         warn(f"{config.configParams.aws.account} is not a valid aws account")
 
-    # aws keypair name
-    if config.configParams.aws.keyPairName is None:
-        config.configParams.aws.keyPairName = getpass.getuser()
-    keyPairName = ask(f"🔑 Key pair name - stored in AWS Sandbox, default [{config.configParams.aws.keyPairName}]: ")
-    if len(keyPairName) > 0:
-        config.configParams.aws.keyPairName = keyPairName
+    # ask user if they want to create a new key or import an existing key
+    if ask_yesno("Do you want to create a new key pair?"):
+        _aws_create_keypair(ctx, config, use_aws_vault=True, aws_account_name=config.configParams.aws.account)
+    elif ask_yesno("Do you want to import an existing key pair?"):
+        _aws_import_keypair(ctx, config, use_aws_vault=True, aws_account_name=config.configParams.aws.account)
+    else:
+        warn("No key pair configured, you will need to manually configure a key pair")
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
You are overriding the stored key pair name, you should keep the previous code and integrate it in this logic. A possible solution would be:
```suggestion
   if config.configParams.aws.keyPairName is None:
   	config.configParams.aws.keyPairName = getpass.getuser()
    # ask user if they want to create a new key or import an existing key
    if ask_yesno("Do you want to create a new key pair?"):
        _aws_create_keypair(ctx, config, use_aws_vault=True, aws_account_name=config.configParams.aws.account)
    elif ask_yesno("Do you want to import an existing key pair?"):
        _aws_import_keypair(ctx, config, use_aws_vault=True, aws_account_name=config.configParams.aws.account)
    else:
        info(f"Using key pair {config.configParams.aws.keyPairName}")
```

And then use `config.configParams.aws.keyPairName` as the default name. I removed the warning as most of the time the key pair is stored in aws agent sandbox

Reply:
It's not being overridden, it's used as the default in [resolve_keypair_opts](https://github.com/DataDog/test-infra-definitions/pull/1045/files#diff-c17ad71d631afbbe74eea29844a37a1501410870263f1ea09d58f4a786e91772R279-R283).

I'll update the warning to only show if no values are set.